tests/e2e/default/enketo/training-cards.wdio-spec.js

```diff
@@ -0,0 +1,89 @@
+const fs = require('fs');
+
+const utils = require('../../../utils');
+const loginPage = require('../../../page-objects/default/login/login.wdio.page');
+const commonPage = require('../../../page-objects/default/common/common.wdio.page');
+const modalPage = require('../../../page-objects/default/common/modal.wdio.page');
+const trainingCardsPage = require('../../../page-objects/default/enketo/training-cards.wdio.page');
+const genericFormPage = require('../../../page-objects/default/enketo/generic-form.wdio.page');
+const placeFactory = require('../../../factories/cht/contacts/place');
+const userFactory = require('../../../factories/cht/users/users');
+const personFactory = require('../../../factories/cht/contacts/person');
+const commonElements = require('../../../page-objects/default/common/common.wdio.page');
+const reportsPage = require('../../../page-objects/default/reports/reports.wdio.page');
+const reportsPo = require('../../../page-objects/default/reports/reports.wdio.page');
+
+describe('Training Cards', () => {
+  const parent = placeFactory.place().build({ _id: 'dist1', type: 'district_hospital' });
+  const user = userFactory.build({ roles: [ 'nurse', 'chw' ] });
+  const patient = personFactory.build({ parent: { _id: user.place._id, parent: { _id: parent._id } } });
+  const formDoc = {
+    _id: 'form:training:text_only',
+    internalId: 'training:text_only',
+    title: 'Text Only Training',
+    type: 'form',
+    context: {
+      start_date: new Date().getTime(),
+      user_roles: [ 'nurse' ],
+      duration: 5,
+    },
+    _attachments: {
+      xml: {
+        content_type: 'application/octet-stream',
+        data: Buffer
+          .from(fs.readFileSync(`${__dirname}/forms/training-cards-text-only.xml`, 'utf8'))
+          .toString('base64'),
+      },
+    },
+  };
+
+  before(async () => {
+    await utils.saveDocs([ parent, patient, formDoc ]);
+    await utils.createUsers([ user ]);
+    await loginPage.login(user);
+    await commonElements.waitForPageLoaded();
+  });
+
+  it('should cancel training', async () => {
+    await commonPage.goToMessages();
+    await trainingCardsPage.waitForTrainingCards();
+    await genericFormPage.cancelForm();
+
+    const modal = await modalPage.getModalDetails();
+    expect(modal.header).to.equal('Important changes');
+    expect(modal.body).to.contain(
+      'This training is not finished. You will lose your progress if you leave now. Are you sure you want to leave?'
+    );
+    await trainingCardsPage.quitTraining();
+
+    await commonPage.goToReports();
+    await commonElements.waitForPageLoaded();
+    await (await reportsPo.noReportSelectedLabel()).waitForDisplayed();
+  });
+
+  it('should complete training', async () => {
+    await commonPage.goToMessages();
+    await commonElements.waitForPageLoaded();
+    browser.refresh();
```

Comment:
Yes, we decided to do it like this because users can quit a training and do it later. We still don't have a page where they can go and see all their pending trainings, and we want to minimize interruptions to their job so that the training won't appear on every navigation and not after X minutes, but it will appear when the app loads. 

They can have X number of pending trainings, but we display 1; after they complete that one, the following training appears on the next load.

We tried this with 2 partners; we didn't receive negative feedback about the interaction. But we can change it later if we heard that they need it. They did ask for a place to see pending training, which we're still considering.

Reply:
About the e2e, the training was displayed during the previous test, so to get it back again I thought of reloading the app.

Reply:
Unfinished training should come back after reload.

Reply:
I've added a comment to this line